src/openforms/prefill/contrib/eidas/tests/test_config_checks.py

```diff
@@ -0,0 +1,114 @@
+
+    def test_no_eidas_OIDC_client_configured(self):
+        plugin = EIDASPrefill(identifier="eidas")
+
+        with self.assertRaises(InvalidPluginConfiguration) as exc:
+            plugin.check_config()
+
+        self.assertEqual(exc.exception.args[0], _("Missing OIDC client for eIDAS."))
+
+    def test_eidas_OIDC_client_disabled(self):
+        OFOIDCClientFactory.create(
+            enabled=False,
+            with_eidas=True,
+            oidc_provider=self.provider,
+        )
+        plugin = EIDASPrefill(identifier="eidas")
+
+        with self.assertRaises(InvalidPluginConfiguration) as exc:
+            plugin.check_config()
+
+        self.assertEqual(exc.exception.args[0], _("Missing OIDC client for eIDAS."))
+
+    def test_eidas_OIDC_client_missing_identity_settings(self):
+        OFOIDCClientFactory.create(
+            enabled=True,
+            with_eidas=True,
+            oidc_provider=self.provider,
+            options__identity_settings=None,
+        )
+        plugin = EIDASPrefill(identifier="eidas")
+
+        # Check the config and expect a raised exception
+        with self.assertRaises(InvalidPluginConfiguration) as exc:
+            plugin.check_config()
+
+        self.assertEqual(
+            exc.exception.args[0],
+            _("Missing OIDC client identity settings for eIDAS."),
+        )
+
+    def test_eidas_OIDC_client_missing_identity_settings_options(self):
+        client = OFOIDCClientFactory.create(
+            enabled=True,
+            with_eidas=True,
+            oidc_provider=self.provider,
+        )
+        plugin = EIDASPrefill(identifier="eidas")
+
+        required_fields = (
+            "legal_subject_identifier_claim_path",
+            "legal_subject_identifier_type_claim_path",
+            "legal_subject_first_name_claim_path",
+            "legal_subject_family_name_claim_path",
+            "legal_subject_date_of_birth_claim_path",
+        )
+
+        # Sanity check; the initial config is correct
+        try:
+            plugin.check_config()
+        except InvalidPluginConfiguration as exc:
+            raise self.failureException("Config check should have passed") from exc
+
+        # Each required_field should trigger an exception, when left empty
+        for field in required_fields:
+            # Clear the field
+            original_data = client.options["identity_settings"][field]
+            client.options["identity_settings"][field] = []
+            client.save()
+
+            # Check the config and expect a raised exception
+            with self.assertRaises(InvalidPluginConfiguration) as exc:
+                plugin.check_config()
+
+            self.assertEqual(
+                exc.exception.args[0],
+                _("Missing OIDC client identity settings for eIDAS."),
+            )
+
+            # Reset the field.
+            client.options["identity_settings"][field] = original_data
+            client.save()
```

Comment:
this is a bit of a smell, because a test failure will lead to this reset not being called, which will affect the other cases

To have guaranteed cleanups, use `self.addCleanup`, but that [doesn't work with subTest](https://github.com/python/cpython/issues/134079) :(